src/solver/pdist/backward_pdist.cpp

```diff
@@ -0,0 +1,284 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <vector>
+
+#include <miopen/miopen.h>
+#include "miopen/conv_solution.hpp"
+#include <miopen/datatype.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/kernel_info.hpp>
+#include <miopen/mlo_internal.hpp>
+#include <miopen/target_properties.hpp>
+#include "miopen/buffer_info.hpp"
+#include "miopen/errors.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/invoke_params.hpp"
+#include "miopen/tensor.hpp"
+#include "miopen/tensor_view_utils.hpp"
+#include <miopen/reduce/utils.hpp>
+
+#include <miopen/pdist.hpp>
+#include <miopen/pdist/solvers.hpp>
+#include <miopen/pdist/invoke_params.hpp>
+#include "miopen/pdist/problem_description.hpp"
+
+#define LOCAL_SIZE 256
+
+namespace miopen {
+
+namespace solver {
+
+namespace pdist {
+
+MultiBufferWorkspaceTraits GetMultiBufferWorkspaceTraits(const TensorDescriptor& inputDesc)
+{
+    auto N = inputDesc.GetLengths()[0];
+    auto M = inputDesc.GetLengths()[1];
+
+    auto input_dtype = inputDesc.GetType();
+
+    return MultiBufferWorkspaceTraits{(N - 1) * N * M * get_data_size(input_dtype)};
+}
+
+bool PdistBackward::IsApplicable(const ExecutionContext& context,
+                                 const miopen::pdist::BackwardProblemDescription& problem) const
+{
+    std::ignore = context;
+
+    if(!problem.IsAllContiguous())
+    {
+        return false;
+    }
+
+    if(!problem.IsAllPacked())
+    {
+        return false;
+    }
+
+    if(!(problem.GetInputDesc().GetType() == miopenFloat ||
+         problem.GetInputDesc().GetType() == miopenHalf ||
+         problem.GetInputDesc().GetType() == miopenBFloat16))
+    {
+        return false;
+    }
+
+    return true;
+}
+
+ConvSolution
+PdistBackward::GetSolution(const ExecutionContext& context,
+                           const miopen::pdist::BackwardProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto dtype = problem.GetInputDesc().GetType();
+
+    auto input_dtype  = miopen::GetDataType(problem.GetdInputDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetdOutputDesc().GetType());
+    auto dinput_dtype = miopen::GetDataType(problem.GetdInputDesc().GetType());
```

Comment:
you have IsSameType so this is redundant

Reply:
I understand that these types are all the same. However, I intentionally query each one separately to enhance code readability and self-documentation. For example, I would like to use `{"OUTPUT_TYPE", dinput_dtype == "bfloat16" ? "ushort" : dinput_dtype}` to explicitly shows that the output type is `dinput_dtype`, which is clearer than using `{"OUTPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype}` (even when I know they are all the same). I think it would be easier for future readers to understand the code.

Pls take your time to review my fix for this and consider if it's ok now. If it's not, pls let me know

Reply:
you can use variable name: dtype then use for all variable, still ensure readability

Reply:
solved:
```
auto dtype = problem.GetInputDesc().GetType();
auto dtype_str = miopen::GetDataType(dtype);
```